### ISSUE TITLE
Add db.close() documentation

### ISSUE THREAD

**Issue by jacargentina** [pouchdb/pouchdb#3429]
I'm i wrong... or pouchdb has a **close** on its API, which is not documented? (at least reading http://pouchdb.com/api.html i cant see it...)

It seems there is a test on test.basics.js !

**Comment by nolanlawson** [comment]
It only really makes sense for IndexedDB; personally I've never used it in any of my apps.

@daleharvey should we deprecate it entirely?

**Comment by nolanlawson** [comment]
> I think its not posible to access the very same pouchdb database from two or more concurrent node processes

This is correct. And I don't think it's really reasonable to put it on the roadmap; either PouchDB has to add a lot of slow, ugly queuing code, or you have to add a lot of ugly code to work with `close()` (assuming `close()` did what you think it does).

Luckily PouchDB/CouchDB were designed from the bottom-up to support the kind of scenario you describe - just have two separate databases and replicate them! :) No need for child processes, worker threads, whatever - just spin up two separate PouchDB Servers and load-balance between them.

Also TBH if you want maximum performance, you're probably going to want to use CouchDB instead of PouchDB Server or `express-pouchdb`.

**Comment by jacargentina** [comment]
I think the child process cant be avoided, as i want my express js "frontend" to continue responsive while the SQL -> pouch takes place (many sql queries running to build docs, to compare against pouch ones, and update differents, and insert missings...)

But now that you told... i'm already using **express-pouchdb** ! So, i'm thinking on using it as a "proxy" to the real pouchdb by doing

**main.js**
```javascript
 var db = new PouchDB('http://localhost/proxy/X');
```

**worker.js**
```javascript
 var db = new PouchDB('X');
 // This will work without problem as main.js is not locking the db on disk!
```

Any thought?

**Comment by nolanlawson** [comment]
Sorry, not sure what you mean?

**Comment by jacargentina** [comment]
Sorry, this is not anymore about close API. Closing.

**Comment by daleharvey** [comment]
> This is correct. And I don't think it's really reasonable to put it on the roadmap; either PouchDB has to
>  add a lot of slow, ugly queuing code, or you have to add a lot of ugly code to work with close() 
> (assuming close() did what you think it does).

We have a serial queue for leveldb, so making that work across process probably isnt a crazy amount of work but yeh getting that right probably isnt going to happen soon, not sure what the state of the art is for cross process leveldb is but I imagine it wont help us a ton since we need transactional integrity.

I believe close was first implemented because we thought it was need to get setup / teardown working in the tests but then managed to do the tests without it, so its got left around. I dont think it is what is needed to have cross process leveldb working though so it probably can be safely removed

**Comment by nolanlawson** [comment]
Let's reopen then to talk about `db.close()`. Removing it would be nice, especially since it's undocumented and apparently unneeded.

**Comment by nolanlawson** [comment]
Actually, closing in favor of https://github.com/pouchdb/pouchdb/issues/3437

**Comment by wxfred** [comment]
> Let's reopen then to talk about `db.close()`. Removing it would be nice, especially since it's undocumented and apparently unneeded.

Why remove it? I'm using indexdb. My electron app creates many projects, each one has its own db under its folder. And users can delete projects. If a db is open, deleting the entire project folder will be blocked by os, because some of the db files are locked.
